src/builder.ts

```diff
@@ -245,15 +273,84 @@ export interface TypegenInfo {
   contextType?: string;
 }
 
+/**
+ * The resolved builder options wherein optional fields have been
+ * given their fallbacks, etc.
+ */
+export interface BuilderConfig extends BuilderOptions {
+  outputs: {
+    schema: false | string;
+    typegen: false | string;
+  };
+}
+
+function resolveBuilderOptions(config: BuilderOptions): BuilderConfig {
+  if (
+    config.outputs !== undefined &&
+    typeof config.outputs !== "boolean" &&
+    typeof config.outputs.schema === "string"
```

Comment:
@tgriesser My revised version is still not safe in JS. This line could still be reached by `null`, in JS. I'm hesitant to starting coding type checks optimized for JS though. If they want great feedback on type errors, don't use JS... For us, I doubt we're  going to have consistent and uniform tight type checks everywhere for JS, so why bother with sprinkles?

Reply:
Yeah, actually `null` was the only thing I was suggesting we guard against, from the original code, it would have been:

```ts
if (
  config.outputs &&
  typeof config.outputs === "object" &&
  typeof config.outputs.schema === "string"
)
```

> For us, I doubt we're going to have consistent airtight type checks everywhere for JS, so why bother with sprinkles of it?

I agree with this, I'm fine with leaving it out - the `null` / `typeof` is one I still generally try to guard against just because it can be a confusing runtime error, `cannot read property schema of null`.